src/trans/internal/suleg_mod.F90

```diff
@@ -874,10 +871,9 @@ SUBROUTINE SULEG
                 ZCLONEA(IMLOC)%COMMSBUF(1:ICLONELEN) = ZRCVBUTFV(1:ICLONELEN,JSETV)
                 CALL UNPACK_BUTTERFLY_STRUCT(S%FA(IMLOC)%YBUT_STRUCT_A,ZCLONEA(IMLOC))
               ENDIF
-              IF(ALLOCATED(ZCLONEA(IMLOC)%COMMSBUF) ) THEN
-                IF( SIZEOF(ZCLONEA(IMLOC)%COMMSBUF) > 0 ) DEALLOCATE(ZCLONEA(IMLOC)%COMMSBUF)
-                !              ZCLONEA(IMLOC)%COMMSBUF=>NULL()
-              ENDIF
+
+              IF( ICLONELEN > 0 ) DEALLOCATE(ZCLONEA(IMLOC)%COMMSBUF)
+
```

Comment:
I think that ICLONELEN could be unininitialized in case
`IF( .NOT.ALLOCATED(ZCLONEA(IMLOC)%COMMSBUF) )THEN` (line 868) is not triggered.
Probably the `IF(ALLOCATED(ZCLONEA(IMLOC)%COMMSBUF) ) THEN` was still correct.
Could it not also be that it needs to stay allocated across iterations? Otherwise why all the guards (line 868).

Reply:
Good point @wdeconinck - we shouldn't use `ICLONELEN`.

The first `IF` has no purpose at all, as far as I can see. I can't see how it's possible for `ZCLONEA(IMLOC)%COMMSBUF` to not be allocated at line 877.

Instead of checking if `ICLONELEN > 0`, why don't we check `STORAGE_SIZE(ZCLONEA(IMLOC)%COMMSBUF) > 0`? `STORAGE_SIZE` is the same as `SIZEOF` but returns the bits allocated rather than bytes, and it appears to be part of the Fortran 2008 standard, unlike `SIZEOF`.